bin/upgrade.bash

```diff
@@ -66,6 +66,13 @@ prepare() {
   for prefix in secrets sc wc; do
     config_validate "${prefix}"
   done
+
+  default_version=$(yq4 '.global.ck8sVersion' "${CK8S_CONFIG_PATH}/defaults/common-config.yaml")
+  override_version=$(yq4 '.global.ck8sVersion // null' "${CK8S_CONFIG_PATH}/common-config.yaml")
+  if [ "$override_version" != null ] && [ "$default_version" != "$override_version" ]; then
+    log_error "ck8sVersion does not match between the common-configs."
```

Comment:
This check should skip if `ck8sVersion: any` and it is better suited to be in the start of `check_version` in `scripts/migration/lib.sh` as that is the main version check used in the upgrade scripts. (And we also get the benefit that the error is printed when trying to run apply.)

Reply:
Got some help @Xartos so the script is different now.

Reply:
I think I'd still like to see a separate check like the one you had above before the regular version checks, since else if you have added a proper version string `vX.Y.Z` then you still get blocked I think (well as long as that version is invalid for apply conditions). And that would make it clear that it because of something that has been manually added to the override config.